docs/Users/Tutorials/Testing_Bears.rst

```diff
@@ -0,0 +1,196 @@
+
+ '''.splitlines(keepends=True)
+
+
+ bad_file = '''
+ """
+ example module level docstring
+ """
+
+ def hello():
+     print("hello world")
+
+ '''.splitlines(keepends=True)
+
+ PyDocStyleBearTest = verify_local_bear(PyDocStyleBear,
+                                        valid_files=(good_file,),
+                                        invalid_files=(bad_file,),
+                                        tempfile_kwargs={"suffix": ".py"})
+
+``good_file`` is a file which follows all the checks that you're bear is
+checking and a ``bad_file`` is one which misses at one or more checks.
+``PyDocStyleBear`` checks the python docstrings.So, you need to write a "good
+file" which follows docstring writing standards and a "bad_file" which doesn't
+and feed them to ``verify_local_bear`` as input along with your bear(PyDocStyle
+Bear in this case) and few additional arguments.That's it !
+
+.. note::
+
+    "good_file" and "bad_file" are lists just like **file** (You haven't gone
+    through the codebase if you don't know what is meant by *file*).
+
+**EXAMPLE 2** using ``LocalBearTestHelper`` class(check_validity method)
+
+*PEP8BearTest*
+
+.. code::
+
+ from queue import Queue
+
+ from bears.python.PEP8Bear import PEP8Bear
+ from tests.LocalBearTestHelper import LocalBearTestHelper
+ from coalib.settings.Section import Section
+ from coalib.settings.Setting import Setting
+
+
+ class PEP8BearTest(LocalBearTestHelper):
+
+     def setUp(self):
+         self.section = Section('name')
+         self.section.append(Setting('max_line_length', '80'))
+         self.uut = PEP8Bear(self.section, Queue())
+
+     def test_valid(self):
+         self.check_validity(self.uut, ["import sys"])
+         self.check_validity(self.uut, ["a = 1 + 1"])
+
+     def test_line_length(self):
+         self.check_validity(self.uut, ["a = 1 + 1 + 1 + 1 + 1 + 1 + 1"])
+         self.section.append(Setting('max_line_length', '10'))
+         self.check_validity(self.uut,
+                             ["a = 1 + 1 + 1 + 1 + 1 + 1 + 1"],
+                             valid=False)
+
+     def test_indent_level(self):
+         test_code = ['def func():\n',
+                      '    pass\n']
+         self.check_validity(self.uut, test_code)
+
+         self.section.append(Setting('tab_width', '2'))
+         self.check_validity(self.uut, test_code, valid=False)
+         self.check_validity(self.uut, ['def func():\n', '  pass\n'])
+
+     def test_disable_warnings(self):
+         test_code = ['def func():\n',
+                      '    pass\n',
+                      'def func2():\n',
+                      '    pass\n']
+         self.check_validity(self.uut, test_code, valid=False)
+
+         self.section.append(Setting('pep_ignore', 'E302'))
+         self.check_validity(self.uut, test_code)
+
+     def test_invalid(self):
+         self.check_validity(self.uut, [""], valid=False)
+         self.check_validity(self.uut, ["a=1+1"], valid=False)
```

Comment:
same issue, you'd like your code sample to be ended here, but i dont think it does.